oat-sa/tao-core
# feat: DI container services availability during install

Ticket - https://oat-sa.atlassian.net/browse/REL-1326

## Goal
DI container services availability during install

## Changelog
- feat: DI container services availability during install
- feat: added the ability to configure the run DI container registered service->method as part extension install process. 

## How to test
Configure extension manifest install section with parameter  ```'containerRebuild' => true,```
Debug one of install->php classes to check the availability of some DI services registered in the provider from ```containerServiceProviders```
Run setup to debug
```docker exec -it studio-phpfpm php tao/scripts/taoSetup.php /var/docker-stack-cli/setup.json -vvv```

To check the ability to run DI container registered service->method configured in the manifest.php file you should add some service to section  [install][php] in this format:
```
            [
                'service' => SomeServiceClass::class,
                'method' => 'methodName',
                'arguments' => ['argument1', ...]
            ]       
```
```containerRebuild``` must be set to true for this extension

Depend on https://github.com/oat-sa/generis/pull/1086
